release/ray_release/job_manager.py

```diff
@@ -0,0 +1,85 @@
+import os
+import time
+
+from typing import Dict, Tuple
+
+from ray_release.logger import logger
+from ray_release.util import ANYSCALE_HOST
+from ray_release.cluster_manager.cluster_manager import ClusterManager
+from ray_release.exception import CommandTimeout
+from ray.job_submission import JobSubmissionClient, JobStatus
+
+
+class JobManager:
+    def __init__(self, cluster_manager: ClusterManager):
+        self.job_id_pool: Dict[int, str] = dict()
+        self.start_time: Dict[int, float] = dict()
+        self.counter = 0
+        self.cluster_manager = cluster_manager
+        self.job_client = None
+        self.last_job_id = None
+
+    def _run_job(self, cmd_to_run, env_vars) -> int:
+        self.counter += 1
+        command_id = self.counter
+        env = os.environ.copy()
+        env["RAY_ADDRESS"] = self.cluster_manager.get_session_name()
+        env.setdefault("ANYSCALE_HOST", ANYSCALE_HOST)
+
+        full_cmd = " ".join(f"{k}={v}" for k, v in env_vars.items()) + " " + cmd_to_run
+        logger.info(f"Executing {cmd_to_run} with {env_vars} via ray job submit")
+
+        job_id = self.job_client.submit_job(
+            # Entrypoint shell command to execute
+            entrypoint=full_cmd,
+        )
+        self.last_job_id = job_id
+        self.job_id_pool[command_id] = job_id
+        self.start_time[command_id] = time.time()
+        return command_id
+
+    def _wait_job(self, command_id: int, timeout: int):
+        start_time = time.monotonic()
+        timeout_at = start_time + timeout
+        next_status = start_time + 30
+
+        while True:
+            now = time.monotonic()
+            if now >= timeout_at:
+                raise CommandTimeout(
+                    f"Cluster command timed out after {timeout} seconds."
+                )
```

Comment:
Should we cancel job execution on Anyscale then? (Probably not urgent as we'll usually shut down the cluster anyway)

Reply:
Do we stop commands for sdk runner? Maybe I can use the same behavior

Reply:
Fine for now, we can do this later